pupil-spa/src/app/services/feedback/feedback.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from '@angular/core';
+import { environment } from '../../../environments/environment';
+import { Http, RequestOptions, Headers } from '@angular/http';
+import { StorageService } from '../storage/storage.service';
+
+@Injectable()
+export class FeedbackService {
+
+  constructor(private http: Http, private storageService: StorageService) {
+  }
+
+  postFeedback(): Promise<any> {
+    return new Promise(async (resolve, reject) => {
+      const headers = new Headers();
+      headers.append('Content-Type', 'application/json');
+      const requestArgs = new RequestOptions({ headers: headers });
+
+      if (!this.storageService.getItem('feedback') || !this.storageService.getItem('access_token')) {
+        return reject('Missing data in local storage');
+      }
+
+      const storedFeedback = this.storageService.getItem('feedback');
```

Comment:
This requires two calls to getItem('feedback') could the defensive check be moved to after the call and operate on the variables?